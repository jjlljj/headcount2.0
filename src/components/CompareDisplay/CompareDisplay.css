.compare-display {
  display: flex;
  justify-content: space-around;
  margin: 20px auto;
  max-width: 786px;
}

.compare-container {
  display: flex;
  justify-content: center;
  min-width: 250px;
  opacity: 0;
  transition: 0.5s;
  visibility: hidden; 
  width: 33.33%;
}

.compare-instructions {
  margin: auto;
  text-align: center;
}

.show {
  opacity: 1;
  visibility: visible; 
}