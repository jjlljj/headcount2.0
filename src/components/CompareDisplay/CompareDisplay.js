import React from 'react';
import { oneOfType, string, func, object } from 'prop-types';
import Card from '../Card/Card';
import CompareCard from '../CompareCard/CompareCard';
import './CompareDisplay.css';

const CompareDisplay = ({ card1, comparison, card2, removeCard }) => {

  return (
    <section className="compare-display"> 
      <div className="compare-container">
        { card1 &&
          <Card card={card1} clickHandler={removeCard} highlight="true"/>
        }
      </div>  
      <div className="compare-container">
        <CompareCard card={comparison} />
      </div>  
      <div className="compare-container">
        { card2 &&
          <Card card={card2} clickHandler={removeCard} highlight="true"/>
        }
      </div>  
    </section>
  );

};

CompareDisplay.propTypes = {
  card1: oneOfType( [object, string] ),
  comparison: oneOfType( [object, string] ),
  card2: oneOfType( [object, string] ),
  removeCard: func
};



export default CompareDisplay;
