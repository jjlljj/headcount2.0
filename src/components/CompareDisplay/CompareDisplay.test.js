/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import CompareDisplay from './CompareDisplay';

describe('CompareDisplay', () => {

  it('should exist', () => {
    const renderedComponent = shallow(<CompareDisplay />)

    expect(renderedComponent).toBeDefined()
  })

  it('should render child components', ()=> {
    const card1 = { location: 'Denver, Co', data: { 2004: 0.007, 2005: 1.112, 200: 7.110 } }
    const card2 = { location: 'Boston, Ma', data: { 2004: 0.023, 2005: 3.112, 2006: 7.110 } }
    const comparison = { "Denver, Co": 4321, "Boston, Ma": 1234, compared: .609 }

    const renderedComponent = shallow(<CompareDisplay card1="card1" card2="card2" comparison="comparison"/>)

    expect(renderedComponent.find('Card')).toBeDefined()
    expect(renderedComponent.find('CompareCard')).toBeDefined()
  })

  it('should match snapshot', ()=> {
    const card1 = { location: 'Denver, Co', data: { 2004: 0.007, 2005: 1.112, 200: 7.110 } }
    const card2 = { location: 'Boston, Ma', data: { 2004: 0.023, 2005: 3.112, 2006: 7.110 } }
    const comparison = { "Denver, Co": 4321, "Boston, Ma": 1234, compared: .609 }

    const renderedComponent = shallow(<CompareDisplay card1="card1" card2="card2" comparison="comparison"/>)

    expect(renderedComponent).toMatchSnapshot()
  });
  
});