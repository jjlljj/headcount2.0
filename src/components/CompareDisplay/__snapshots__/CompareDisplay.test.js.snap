// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareDisplay should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareDisplay
    card1="card1"
    card2="card2"
    comparison="comparison"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="compare-container"
>
          <Card
                    card="card1"
                    clickHandler={undefined}
                    highlight="true"
          />
</div>,
        <div
          className="compare-container"
>
          <CompareCard
                    card="comparison"
          />
</div>,
        <div
          className="compare-container"
>
          <Card
                    card="card2"
                    clickHandler={undefined}
                    highlight="true"
          />
</div>,
      ],
      "className": "compare-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Card
            card="card1"
            clickHandler={undefined}
            highlight="true"
/>,
          "className": "compare-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "card": "card1",
            "clickHandler": undefined,
            "highlight": "true",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CompareCard
            card="comparison"
/>,
          "className": "compare-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "card": "comparison",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Card
            card="card2"
            clickHandler={undefined}
            highlight="true"
/>,
          "className": "compare-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "card": "card2",
            "clickHandler": undefined,
            "highlight": "true",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="compare-container"
>
            <Card
                        card="card1"
                        clickHandler={undefined}
                        highlight="true"
            />
</div>,
          <div
            className="compare-container"
>
            <CompareCard
                        card="comparison"
            />
</div>,
          <div
            className="compare-container"
>
            <Card
                        card="card2"
                        clickHandler={undefined}
                        highlight="true"
            />
</div>,
        ],
        "className": "compare-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Card
              card="card1"
              clickHandler={undefined}
              highlight="true"
/>,
            "className": "compare-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "card": "card1",
              "clickHandler": undefined,
              "highlight": "true",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CompareCard
              card="comparison"
/>,
            "className": "compare-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "card": "comparison",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Card
              card="card2"
              clickHandler={undefined}
              highlight="true"
/>,
            "className": "compare-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "card": "card2",
              "clickHandler": undefined,
              "highlight": "true",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
